### PR TITLE
feat(CEECORE-3011) add a basic react v18 based subapp version 2Subapp2

### PR DESCRIPTION
**Pull request by mristic505**
### Description

- This SubApp is located in `sample/subapp2-baisc` directory.
- It demonstrates basic `ServerSide rendering`, and `Redux implementation` with mock data  features and `Routing`
- It has two routes:
1.  `/` Home 
2.  `/products` 

### Completed
- [x] Add basic react v18 based SubApp that uses Version 2 packages.
- [x] Upgrade Version 2 packages
- [x] Demonstrate Redux implementation
- [x] implement react-router feature

### CODE REVIEW

File: packages/xarc-react-router/package.json [hunk 1]
```diff
@@ -48,7 +48,7 @@
     "eslint": "^8.16.0",
     "eslint-config-walmart": "^2.2.1",
     "eslint-plugin-filenames": "^1.3.2",
-    "eslint-plugin-jsdoc": "^39.3.2",
+    "eslint-plugin-jsdoc": "^32.0.2",
```
Comment (smuthya) [inline packages/xarc-react-router/package.json:51]: is there specific reason you referred to lower version here

File: samples/subapp2-basic/package.json [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    "@xarc/react": "../../packages/xarc-react",
+    "@xarc/react-query": "../../packages/xarc-react-query",
+    "@xarc/react-redux": "../../packages/xarc-react-redux",
+    "@xarc/react-router": "../../packages/xarc-react-router"
+  },
+  "devDependencies": {
+    "@types/node": "^14.14.6",
+    "@xarc/app-dev": "../../packages/xarc-app-dev",
+    "@xarc/run": "1.1.1",
+    "@xarc/opt-postcss": "../../packages/xarc-opt-postcss",
```
Comment (smuthya) [inline samples/subapp2-basic/package.json:47]: can we follow same pattern as in other samples and add fyn dependencies with relative url

**Review by smuthya — APPROVED** [review]
apart from minor changes, rest all looks good

File: packages/xarc-react-router/package.json (hunk 1 above)
Comment (mristic505) [reply packages/xarc-react-router/package.json:51]: At one point 39.3.2 was working fine then it started breaking out of nowhere and now it's working fine again. I'll change it back to 39.3.2.

**Review by divyakarippath — APPROVED** [review]

**Pull request merged**